src/network/wait-online/manager.c

```diff
@@ -352,6 +354,13 @@ static int on_dns_configuration_event(
         assert(link);
 
         if (error_id) {
+                if (streq(error_id, SD_VARLINK_ERROR_DISCONNECTED)) {
+                        log_debug("DNS configuration monitor disconnected, reconnecting...");
+
+                        m->varlink_client = sd_varlink_unref(m->varlink_client);
+                        return manager_dns_configuration_listen(m);
```

Comment:
hmm, manager_dns_configuration_listen() doesn't log on its own, hence you should do here, so that any error is visible.